### PR TITLE
Increasing memory and CPU limits for core/critical components

### PR DESCRIPTION
**Pull request by ggaurav10**
Increasing memory and CPU limits for kube-proxy, ingress-controller, calico and etcd based on observations from performance tests.

**What this PR does / why we need it**:
For network heavy workloads, the limits assigned to kube-proxy, ingress controller weren't enough. Thus need to be increased.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
The "requests" values are kept the same, and might be changed as part of another PR.
Idea is to launch pods so that they consume/reserve less resources to begin with. The limits are increased so that those critical components can grow if necessary.

**Release note**:

```improvement operator

```

### CODE REVIEW

File: charts/shoot-cloud-config/charts/original/templates/kubelet/_kubelet-config-file
Comment (mvladev) [inline charts/shoot-cloud-config/charts/original/templates/kubelet/_kubelet-config-file]: I would rather change this to a percentage - this is a hard punishment for a machine with low base memory.

**Review by mvladev — CHANGES_REQUESTED** [review]
This should also be changed
https://github.com/gardener/gardener/blob/6d8de051b67b470f7a6eab9102dc961d79fa9741/charts/shoot-cloud-config/charts/original/templates/kubelet/_kubelet.flags

File: charts/shoot-cloud-config/charts/original/templates/kubelet/_kubelet-config-file
Comment (ggaurav10) [reply charts/shoot-cloud-config/charts/original/templates/kubelet/_kubelet-config-file]: i haven't done experiments with concrete numbers, however, picked up these numbers from the examples in documentation. The example talked about 10% or 1GB for a machine with 10GB of memory. Keeping 5% or 10% for the machines of sizes 60GB or 120GB, and reserving so much memory without any experiments doesn't seem very good.
WDYT? Do you have any recommendations on how to proceed with percentages for different sizes of machines, or can we go with the current implementation

Comment (mvladev) [reply charts/shoot-cloud-config/charts/original/templates/kubelet/_kubelet-config-file]: I'll rather have percentage for a machine with low memory - let's say 10-15gb and after that a static number.

For machine with `< 8Gi` memory (threshold value is just an example) we might put `10%` for soft and `5%` for hard.
For machine with `> 8Gi` memory (threshold value is just an example) we might put `1Gi` for soft and `1.5Gi` for hard.

Comment (ggaurav10) [reply charts/shoot-cloud-config/charts/original/templates/kubelet/_kubelet-config-file]: Thanks @mvladev for suggestions. I will submit the change involving eviction thresholds separately in that case. This will require right design such that the memory size can be queried for machines from different cloud provider (possibly different OS...?), and the values can be configured accordingly. Right now the values are static.

Comment (mvladev) [reply charts/shoot-cloud-config/charts/original/templates/kubelet/_kubelet-config-file]: This information is already available on the CloudProvider https://github.com/gardener/gardener/blob/6d8de051b67b470f7a6eab9102dc961d79fa9741/example/30-cloudprofile-aws.yaml#L35-L39

Comment (ggaurav10) [reply charts/shoot-cloud-config/charts/original/templates/kubelet/_kubelet-config-file]: Thanks for the pointer @mvladev. I will raise another PR for the eviction threshold changes

**Comment by vlerenc** [comment]
@ggaurav10 I thought you would also increase the kubelet eviction settings (that's, I guess, what @mvladev also refered to)? Otherwise the next test will again require manual tweaking. Or is that a separate PR?

@rfranzke Can you and the team make a patch release and get it to dev for testing, because most of you will be out of office next week?

**Comment by rfranzke** [comment]
Let's first evaluate whether these settings are fine/work at all. We can deploy this PR and wait for the test results before it gets merged.

**Comment by ggaurav10** [comment]
@vlerenc: I will be raising a separate PR for eviction policy. However, below is a request from Yariv, who is running the performance tests:
> We have some concerns with the eviction threshold that will not be part of the change 
(Today we experienced a problem that may relate to kubelet and eviction threshold in our canary cluster)
> Therefore I recommend that in the dev landscape test the eviction threshold will use the default value to examine the real behavior in canary and production.

Should we wait until we deploy eviction threshold related changes?

@rfranzke: These numbers are decided based on data collected when Yariv and the team ran performance tests. As discussed on call, i will raise the PR again on separate branch on gardener so that it can be deployed without merging.

**Comment by rfranzke** [comment]
Closed in favour of #592, please follow up discussions there.

**Comment by ggaurav10** [comment]
#593 PR for change in eviction thresholds dynamically based on machine size

**Pull request closed without merging**
